### DISCUSSION TITLE
Support for a non-interactive network graph for static reporting

### DISCUSSION THREAD

**Discussion by mattwarkentin** [ropensci/targets#1071] [Help]
Hi @wlandau,

Would you consider supporting a function like `tar_visnetwork()` that produces a non-interactive graph (`ggraph`??) that can be easily included in DOCX and PDF reports? Often I want to include the network in reports to walk clients through an analysis, but sometimes/usually the report has to be non-HTML.

I know that the network data itself can be obtained with `tar_network()` so the user has the option to create their own, but I thought it would be worthwhile to have an internal option to create a non-interactive graph that looks aesthetically similar to the one seen with `tar_visnetwork()` for consistency. What do you think?

**Comment by ginolhac** [comment]
Hello,

What about `tar_mermaid()` ?
Works great in Quarto docs.

**Comment by mattwarkentin** [comment]
It was my understanding that `tar_mermaid()` relies on a JavaScript library that wouldn't be supported in non-HTML documents. Is this not the case?

**Comment by ginolhac** [comment]
True but it can then rendered as you like it best.
Otherwise there is webshot to convert visnetwork to a png.

**Comment by wlandau** [comment]
`tar_mermaid()` itself does not rely on JavaScript because it only outputs the text required to produce a mermaid.js graph. As @ginolhac said, Quarto can render that text, possibly in output formats like pdf and docx.

To render the `tar_mermaid()` text into a Quarto document, you can programmatically write a `{mermaid}` code chunk like this:

    ```{r, results = "asis", echo = FALSE}
    cat(c("```{mermaid}", tar_mermaid(), "```"), sep = "\n")
    ```

Other alternatives like `DiagrammeR` may have other components like JavaScript, c.f. https://github.com/rich-iannone/DiagrammeR/issues/457 and https://github.com/rich-iannone/DiagrammeR/issues/475.

**Reply by mattwarkentin** [reply]
Right, it doesn't require JS to produce the text output but it needs JS to render the mermaid plot. I am not sure it makes sense to render the mermaid text for an output format (PDF/LaTeX/DOCX) where JS isn't supported. This is why I am interested in a non-web based visual output such as one made with `{ggplot2}` or one of the extension packages like `{ggraph}` for networks. PDF and Word output formats are very common and there is no existing solution to visualize a `targets` pipeline in these documents without workarounds.

For the record, I am using Quarto for rendering to PDF and DOCX.
